sql/2020/11_Security/feature_adoption_by_technology.sql

```diff
@@ -0,0 +1,43 @@
+#standardSQL
+# Adoption of features based on the technology that is used
+SELECT
+  client,
+  category,
+  app,
+  headername,
+  COUNT(DISTINCT url) AS total_pages_with_technology,
+  COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL)) AS total_https_pages,
+  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername)), url, NULL)) AS freq,
+  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername)), url, NULL)) / COUNT(DISTINCT url) AS pct,
+  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername)) AND STARTS_WITH(url, 'https'), url, NULL)) / COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL)) AS pct_https
```

Comment:
Add space to avoid CSP and CSP-RO, and switch to SAFE_DIVIDE

```suggestion
  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)) AS freq,
  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)), COUNT(DISTINCT url)) AS pct,
  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')) AND STARTS_WITH(url, 'https'), url, NULL)), COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL))) AS pct_https
```